psydac/api/ast/nodes.py

```diff
@@ -1831,3 +2197,31 @@ def construct_itergener(a, index):
 
     return iterator, generator
 
+#=============================================================================================
+# the Expression class, works with fixed dimension expressions instead of vectorized one,
+# where in some cases we need to treat each dimesion diffrently
```

Comment:
What do you mean by "fixed dimension expressions"? An expression that applies to a specific scalar entry of an array, rather than the whole array?

Reply:
not only that, much more general, we can create any instruction for a specific dimension.
for example, we have the class `Loop`, this class generates 2 for loops in 2D and 3 in 3D,  the expressions that are generated are the same for 2D and 3D, because they are written in a way that allows them to be applied in any dimension, with the fixed dimension expression we can specify the generated code for a specific dimension, so the generated code in the second dimension of the 2D loop is diffrent from the one in the first dimension of the 2D loop

Reply:
That looks very useful and important. It definitely needs to be explained in the docstring

Reply:
done in [292d40d](https://github.com/pyccel/psydac/pull/156/commits/292d40d572907586fc3ccfe0c1b8b32b64ca19b2)

Reply:
done